modules/library/render/src/main/java/org/geotools/renderer/style/SLDStyleFactory.java

```diff
@@ -1036,6 +1028,30 @@ Stroke getStroke(org.geotools.styling.Stroke stroke, Object feature) {
 		return stroke2d;
 	}
 
+    public static float[] evaluateDashArray(org.geotools.styling.Stroke stroke, Object feature) throws NumberFormatException {
+        float[] dashes = null;
+        if(stroke.dashArray() != null) {
+            if (stroke.dashArray().size()==1) {
+                String dashString = stroke.dashArray().get(0).evaluate(feature, String.class);
+                if (dashString!=null && !"".equals(dashString)) {
+                    String[] strok = dashString.split(" ");
+                    dashes = new float[strok.length];
+                    for (int l = 0; l < dashes.length; l++) {
+                        dashes[l] = Float.parseFloat(strok[l]);
+                    }
+                }
+            } else {
+                dashes = new float[stroke.dashArray().size()];
+                int index = 0;
+                for (Expression expression : stroke.dashArray()) {
+                    dashes[index] = expression.evaluate(feature, Float.class);
+                    index++;
+                }
+            }
+        }
+        return dashes;
+    }
```

Comment:
Sure one could generalize this method. My problem is performance. The original way should be faster. Thats why I decided to make it a special case. But if you prefer to process all possibilities like so I will change it.